### PR TITLE
544 nyc signature algorithm

### PR DESCRIPTION
**Pull request by allthesignals**
Addresses #544 by adding the NYC.ID authentication signature algorithm.

Also fixes many tests. Currently trying to find best way around the required environmental configuration for CRM... (hence why CLI tests fail)

### CODE REVIEW

File: queries/sql/projects-all-ids.js [hunk 1]
```diff
@@ -73,7 +73,7 @@ function publicStatusFilter(publicStatus) {
 }
 
 function certifiedReferredFilter(certifiedReferred) {
-  if (!certifiedReferred.length > 1) return '';
+  if (!certifiedReferred.length) return '';
```
Comment (ghost) [inline queries/sql/projects-all-ids.js]: TDLR this should be:
```
if (!(certifiedReferred.length > 1)) return '';
```

long read: 
ok so upon looking at this again, the original condition is correct. The problem is just that the conditional check needs to be wrapped in () to ensure that it's all evaluated correctly. In the previous state I believe it ends up being: 
!certifiedReferred.length > 1
!0 > 1
true > 1 
false

corrected would be:
!(certifiedReferred.length > 1)
!(0 > 1)
!(false)
true

File: scripts/trigger-update-geometries.js [hunk 2]
```diff
@@ -37,6 +37,6 @@ async function triggerUpdateGeometriesTask() {
   setTimeout(triggerUpdateGeometriesTask, intervalMilli);
 }
 
-//slack.send(`Initializing newly filed project update task with ${INTERVAL_MIN} min interval`);
+// slack.send(`Initializing newly filed project update task with ${INTERVAL_MIN} min interval`);
 console.log('foobar');
```
Comment (ghost) [inline scripts/trigger-update-geometries.js]: while you're already in here, can you delete line 41 and uncomment line 40?

File: queries/sql/projects-all-ids.js (hunk 1 above)
Comment (allthesignals) [reply queries/sql/projects-all-ids.js]: 0 is falsey and any other value evals to truthy (array lengths can't be negative). so `!certifiedReferred.length` should be fine, no?

**Review by ghost — APPROVED** [review]
added two small comments: one fix and one cleanup ask; approved but I think the fix should be implemented.

File: queries/sql/projects-all-ids.js (hunk 1 above)
Comment (ghost) [reply queries/sql/projects-all-ids.js]: but if it has only one thing in it, that's not valid. I'm checking that it has MORE THAN one element, ebcause in the templated query we reference certifiedReferred[0] and certifiedReferred[1]

Comment (allthesignals) [reply queries/sql/projects-all-ids.js]: aaah gotcha. thanks

**Pull request merged**